src/App.js

```diff
@@ -16,6 +16,7 @@ export default class App extends React.Component {
     searchUsers(event) {
         const myEl = event.target.value;
         setTimeout(() => {
+            console.log('Run search handler here with debounce, does not work as expected');
```

Comment:
The debounce will not run OK, here, easy to see that just by putting a console log. Debounce means that at every key stroke the 300ms timeout will fire again (like you did), and the old one will be canceled. So I can type the entire name (usually a user types with delay < 300ms between key strokes).